libnavigator/src/main/java/com/mapbox/navigation/navigator/internal/MapboxNativeNavigatorImpl.kt

```diff
@@ -154,12 +155,19 @@ object MapboxNativeNavigatorImpl : MapboxNativeNavigator {
 
     // Routing
 
-    override suspend fun setPrimaryRoute(
-        routeWithStartingLeg: Pair<NavigationRoute, Int>?
-    ): RouteInfo? = suspendCancellableCoroutine { continuation ->
-        navigator!!.setPrimaryRoute(
-            routeWithStartingLeg?.first?.nativeRoute(),
-            routeWithStartingLeg?.second ?: 0
+    override suspend fun setRoutes(
+        primaryRoute: NavigationRoute?,
+        startingLeg: Int,
+        alternatives: List<NavigationRoute>,
```

Comment:
```suggestion
    internal class RoutesWrapper(
        val primaryRoute: NavigationRoute,
        val startingLeg: Int,
        val alternatives: List<NavigationRoute>
    )
    
    override suspend fun setRoutes(
        routesWrapper: RoutesWrapper?
    )
```

Reply:
tried it, but it makes usage of mocks more complex

Reply:
need to wrap your mocks to `RoutesWrapper` or mock `RoutesWrapper` as well. Do you mean another complex?

Reply:
I'm saying that it's easier to test when I have 3 separate parameters instead of an `object`. Can you please rewrite following verify for object case:
```
verify {
    navigator.setRoutes(any(), any(), eq(listOf(alternativeRoute))
}

Reply:
you need to catch value using [capturing+slot](https://mockk.io/#capturing) and compare it via `asserts`. verify can be replaced by 
```
verify(exactly = 1) { 
  navigator.setRoutes(any(), any(), any())
}
```
or `verify(ordering = ...)` depends on purposes 

you can find samples in nav SDK tests

Reply:
I find current option easier and more convenient. Do you think it's important to apply suggested change?

Reply:
I suppose API becomes less readable when the SDK can send `null` route with legIndex and a list of alternatives. It looks like worsening existing API